### PR TITLE
Betting UI - Back/Lay button

### PR DESCRIPTION
**Pull request by rubyweb3**
https://github.com/AugurProject/augur/issues/6149

Some example:
Screen Shot 2020-03-02 at 11 49 42 AM
Screen Shot 2020-03-02 at 11 46 51 AM

### CODE REVIEW

File: packages/augur-ui/src/modules/common/buttons.tsx [hunk 1]
```diff
@@ -559,6 +559,34 @@ interface EtherscanLinkTSXProps {
...
 }
 
+export const BettingLayButton = (props: DefaultButtonProps) => (
+  <button
+    onClick={e => props.action(e)}
+    className={classNames(Styles.BettingLayButton)}
+    disabled={props.disabled}
+    title={props.title || props.text}
+  >
+    <div>
```
Comment (JohnDanz) [inline packages/augur-ui/src/modules/common/buttons.tsx]: Do we need this extra div?

File: packages/augur-ui/src/modules/common/buttons.tsx [hunk 2]
```diff
@@ -559,6 +559,34 @@ interface EtherscanLinkTSXProps {
   showNonLink?: boolean;
 }
 
+export const BettingLayButton = (props: DefaultButtonProps) => (
```
Comment (JohnDanz) [inline packages/augur-ui/src/modules/common/buttons.tsx]: destruct the props here please:

```
{
 action,
 title
etc...
}
```

File: packages/augur-ui/src/modules/common/buttons.styles.less [hunk 3]
```diff
@@ -1004,6 +1004,90 @@
...
 
+.BettingLayButton {
+  .Button;
+
+  background: transparent;
+  border: @size-1 solid var(--color-outcome-five);
+  color: var(--color-primary-text);
+  padding: @size-6 @size-12;
+
+  > div {
```
Comment (rubyweb3) [inline packages/augur-ui/src/modules/common/buttons.styles.less]: remove div

File: packages/augur-ui/src/modules/common/buttons.tsx [hunk 4]
```diff
@@ -559,6 +559,34 @@ interface EtherscanLinkTSXProps {
...
+    title={props.title || props.text}
+  >
+    <div>
+      <div>{props.text}</div>
+      <div>{props.subText}</div>
+    </div>
+  </button>
+);
+
+export const BettingBackButton = (props: DefaultButtonProps) => (
```
Comment (rubyweb3) [inline packages/augur-ui/src/modules/common/buttons.tsx]: Merge two buttons

File: packages/augur-ui/src/modules/common/buttons.styles.less [hunk 5]
```diff
@@ -1004,6 +1004,90 @@
...
+      .text-12-bold;
+
+      color: var(--color-primary-text);
+      text-align: center;
+    }
+
+    > div:last-of-type {
+      .text-12;
+
+      color: var(--color-secondary-text);
```
Comment (rubyweb3) [inline packages/augur-ui/src/modules/common/buttons.styles.less]: color is not correct

File: packages/augur-ui/src/modules/common/buttons.styles.less (hunk 3 above)
Comment (rubyweb3) [reply packages/augur-ui/src/modules/common/buttons.styles.less]: Updated

File: packages/augur-ui/src/modules/common/buttons.tsx (hunk 4 above)
Comment (rubyweb3) [reply packages/augur-ui/src/modules/common/buttons.tsx]: Updated

File: packages/augur-ui/src/modules/common/buttons.styles.less (hunk 5 above)
Comment (rubyweb3) [reply packages/augur-ui/src/modules/common/buttons.styles.less]: Updated

File: packages/augur-ui/src/modules/common/buttons.tsx (hunk 1 above)
Comment (rubyweb3) [reply packages/augur-ui/src/modules/common/buttons.tsx]: Removed

File: packages/augur-ui/src/modules/common/buttons.tsx (hunk 2 above)
Comment (rubyweb3) [reply packages/augur-ui/src/modules/common/buttons.tsx]: Updated

File: packages/augur-ui/src/modules/common/constants.ts [hunk 6]
```diff
@@ -683,6 +683,8 @@ export const WARNING = 'warning';
 export const ERROR = 'error';
 export const UP = 'up';
 export const DOWN = 'down';
+export const BETTING_LAY = 'betting lay';
```
Comment (JohnDanz) [inline packages/augur-ui/src/modules/common/constants.ts]: if these aren't used for text, we shouldn't use spaces in it. in other words if it's just for setting a type (back or lay) we should just have it match the const name `BETTING_LAY = 'BETTING_LAY';` kind of deal

File: packages/augur-ui/src/modules/common/buttons.styles.less [hunk 7]
```diff
@@ -1006,6 +1006,70 @@
...
+
+  > div:last-of-type {
+    .text-12;
+
+    color: var(--color-primary-text);
+    text-align: center;
+  }
+
+  &.Back {
+    border: @size-1 solid var(--color-outcome-five);
```
Comment (JohnDanz) [inline packages/augur-ui/src/modules/common/buttons.styles.less]: these colors are incorrect, according to the design color chart outcome 5 is actually `#1B91FF` while the betting for button uses: `#5A88EE`  Use `--color-positive` instead.

File: packages/augur-ui/src/modules/common/buttons.styles.less [hunk 8]
```diff
@@ -1006,6 +1006,70 @@
...
+
+    &:disabled {
+      background: var(--color-inactive-text);
```
Comment (JohnDanz) [inline packages/augur-ui/src/modules/common/buttons.styles.less]: this should be `background: var(--color-positive-15);` with no opacity defined.

File: packages/augur-ui/src/modules/common/buttons.styles.less [hunk 9]
```diff
@@ -1006,6 +1006,70 @@
...
+      opacity: 0.15;
+    }
+  
+    &:hover:enabled {
+      background: var(--color-outcome-five-40);
```
Comment (JohnDanz) [inline packages/augur-ui/src/modules/common/buttons.styles.less]: use `--color-positive-40` here instead and all other references that are similar.

File: packages/augur-ui/src/modules/common/buttons.styles.less [hunk 10]
```diff
@@ -1006,6 +1006,70 @@
...
+    }
+  
+    &:active:enabled {
+      background: var(--color-outcome-five-40);
+    }
+  }
+
+  &.Lay {
+    border: @size-1 solid var(--color-outcome-two);
```
Comment (JohnDanz) [inline packages/augur-ui/src/modules/common/buttons.styles.less]: this should be `--color-negative`

File: packages/augur-ui/src/modules/common/buttons.styles.less [hunk 11]
```diff
@@ -1006,6 +1006,70 @@
...
+
+    &:disabled {
+      background: var(--color-inactive-text);
```
Comment (JohnDanz) [inline packages/augur-ui/src/modules/common/buttons.styles.less]: `--color-negative-15` and no opacity change

File: packages/augur-ui/src/modules/common/buttons.styles.less [hunk 12]
```diff
@@ -1006,6 +1006,70 @@
...
+      opacity: 0.15;
+    }
+  
+    &:hover:enabled {
+      background: var(--color-outcome-two-40);
```
Comment (JohnDanz) [inline packages/augur-ui/src/modules/common/buttons.styles.less]: `--color-negaitve-40`

File: packages/augur-ui/src/modules/common/buttons.tsx [hunk 13]
```diff
@@ -552,6 +558,30 @@ export const FilterButton = (props: DefaultActionButtonProps) => (
...
+export interface BettingBackLayButtonProps {
+  title?: string;
+  text?: string;
+  subText?: string;
+  action: Function;
+  disabled?: boolean;
+  type: BETTING_LAY | BETTING_BACK;
+}
+
+export const BettingBackLayButton = (props: BettingBackLayButtonProps) => (
```
Comment (JohnDanz) [inline packages/augur-ui/src/modules/common/buttons.tsx]: deconstruct props

**Review by JohnDanz — CHANGES_REQUESTED** [review]

File: packages/augur-ui/src/modules/common/buttons.tsx (hunk 13 above)
Comment (rubyweb3) [reply packages/augur-ui/src/modules/common/buttons.tsx]: What is deconstruct props mean? Could you give me an example in the codebase?

Comment (rubyweb3) [reply packages/augur-ui/src/modules/common/buttons.tsx]: I think I got it.

File: packages/augur-ui/src/modules/common/constants.ts (hunk 6 above)
Comment (rubyweb3) [reply packages/augur-ui/src/modules/common/constants.ts]: Updated. Thank you.

File: packages/augur-ui/src/modules/common/buttons.styles.less (hunk 7 above)
Comment (rubyweb3) [reply packages/augur-ui/src/modules/common/buttons.styles.less]: Updated. Thank you.

File: packages/augur-ui/src/modules/common/buttons.styles.less (hunk 8 above)
Comment (rubyweb3) [reply packages/augur-ui/src/modules/common/buttons.styles.less]: Updated. Thank you.

File: packages/augur-ui/src/modules/common/buttons.styles.less (hunk 9 above)
Comment (rubyweb3) [reply packages/augur-ui/src/modules/common/buttons.styles.less]: Updated. Thank you.

File: packages/augur-ui/src/modules/common/buttons.styles.less (hunk 10 above)
Comment (rubyweb3) [reply packages/augur-ui/src/modules/common/buttons.styles.less]: Updated. Thank you.

File: packages/augur-ui/src/modules/common/buttons.styles.less (hunk 11 above)
Comment (rubyweb3) [reply packages/augur-ui/src/modules/common/buttons.styles.less]: Updated. Thank you.

File: packages/augur-ui/src/modules/common/buttons.tsx (hunk 13 above)
Comment (rubyweb3) [reply packages/augur-ui/src/modules/common/buttons.tsx]: Updated. Thank you.

**Review by JohnDanz — APPROVED** [review]

**Pull request merged**